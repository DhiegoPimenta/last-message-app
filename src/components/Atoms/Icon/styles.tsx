import styled, { css } from 'styled-components'
import { IconProps } from '.'

const wrapperModifiers = {
    small: () => css`
        width: 3.5rem;
        height: 3.5rem;
    `,
    medium: () => css`
        width: 7.25rem;
        height: 7.25rem;
    `,
    large: () => css`
        width: 11rem;
        height: 11rem;
    `,
}

export const Image = styled.img<Pick<IconProps, 'size'>>`
    ${({ size }) => css`
        ${wrapperModifiers[size!]()}
    `}
`
