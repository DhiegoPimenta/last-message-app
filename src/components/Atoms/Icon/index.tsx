import * as S from './styles'
export type IconsNames =
    | 'document'
    | 'home'
    | 'mail'
    | 'phone'
    | 'trash'
    | 'unlock'
    | 'user'
    | 'send'

export type DataIcon = {
    label: string
    image: string
}

export type IconsDataProps = Record<IconsNames, DataIcon>

export type IconProps = {
    iconName: IconsNames
    size?: 'small' | 'medium' | 'large'
    handleClick?: () => void
}

const Icon = ({ iconName, size = 'medium', handleClick }: IconProps) => {
    const iconsData: IconsDataProps = {
        document: {
            label: 'Icone de documento',
            image: '/img/icons/document.png',
        },
        home: { label: 'Icone de casa', image: '/img/icons/home.png' },
        mail: { label: 'Icone de e-mail', image: '/img/icons/mail.png' },
        phone: { label: 'Icone de telefone', image: '/img/icons/phone.png' },
        trash: { label: 'Icone de lixeira', image: '/img/icons/trash.png' },
        unlock: { label: 'Icone de cadeado', image: '/img/icons/unlock.png' },
        user: { label: 'Icone de usu√°rio', image: '/img/icons/user.png' },
        send: {
            label: 'Icone de envio de mensagem',
            image: '/img/icons/send.png',
        },
    }
    return (
        <S.Image
            size={size}
            aria-label={iconsData[iconName].label}
            src={iconsData[iconName].image}
            onClick={handleClick}
        />
    )
}

export default Icon
