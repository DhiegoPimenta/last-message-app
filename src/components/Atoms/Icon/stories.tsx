import { Story, Meta } from '@storybook/react/types-6-0'

import Icon, { IconProps } from '.'

export default {
    title: 'Atoms/Icon',
    component: Icon,
} as Meta

export const Basic: Story<IconProps> = (args) => <Icon {...args} />
Basic.args = {
    iconName: 'mail',
    size: 'medium',
}
