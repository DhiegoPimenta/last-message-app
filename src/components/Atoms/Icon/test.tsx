import { fireEvent, screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Icon from '.'

describe('<Icon/>', () => {
    it('Should be render Icon', () => {
        const { container } = renderWithTheme(<Icon iconName="home" />)

        expect(screen.getByLabelText('Icone de casa')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be render bigger', () => {
        renderWithTheme(<Icon iconName="home" size="large" />)

        expect(screen.getByLabelText('Icone de casa')).toHaveStyle({
            width: '11rem',
            height: '11rem',
        })
    })

    it('Should be render smaller', () => {
        renderWithTheme(<Icon iconName="home" size="small" />)

        expect(screen.getByLabelText('Icone de casa')).toHaveStyle({
            width: '3.5rem',
            height: '3.5rem',
        })
    })

    it('Should be exec handle on click', () => {
        const mockClick = jest.fn()

        renderWithTheme(
            <Icon iconName="home" size="small" handleClick={mockClick} />
        )

        fireEvent.click(screen.getByLabelText('Icone de casa'))

        expect(mockClick).toHaveBeenCalled()
    })
})
