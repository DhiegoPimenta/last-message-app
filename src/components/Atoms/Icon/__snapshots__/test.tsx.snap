// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> Should be render Icon 1`] = `
.c0 {
  width: 7.25rem;
  height: 7.25rem;
}

<img
  aria-label="Icone de casa"
  class="c0"
  src="/img/icons/home.png"
/>
`;
